docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,497 @@
+              hostPath:
+                path: /etc/cni/net.d
+            - name: xtables-lock
+              hostPath:
+                path: /run/xtables.lock
+                type: FileOrCreate
+            - name: lib-modules
+              hostPath:
+                path: /lib/modules
+---
+apiVersion: addons.cluster.x-k8s.io/v1alpha4
+kind: ClusterResourceSet
+metadata:
+  name: cni
+spec:
+  strategy: ApplyOnce
+  clusterSelector:
+    matchLabels:
+      cni: kindnet
+  resources:
+    - name: kindnet
+      kind: ConfigMap
+```
+
+Copy the above yaml into a local text file - kindnet.yaml - and run:
+```bash
+kubectl apply -f kindnet.yaml
+```
+
+#### Apply the workload cluster
+The below is a Cluster definition that leverages the ClusterClass created above to define its shape.
+```yaml
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: Cluster
+metadata:
+  name: "clusterclass-quickstart"
+  namespace: default
+  labels:
+    cni: kindnet
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.128.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  topology:
+    class: cluster-class
+    version: v1.21.2
+    controlPlane:
+      metadata:
+        labels:
+          topologyLabel: topologyLabelValue
+        annotations:
+          topologyAnnotation: topologyAnnotation
+      replicas: 1 
+    workers:
+      machineDeployments:
+        - class: linux-worker
+          name: linux-workers
+          metadata:
+            labels:
+              topologyLabel: topologyLabelValue
+            annotations:
+              topologyAnnotation: topologyAnnotation
+          replicas: 1 
+```
+
+To create the cluster save the above yaml in a local file - clusterclass-quickstart.yaml - and run:
+
+```bash
+kubectl apply -f clusterclass-quickstart.yaml
+```
+#### Accessing the workload cluster
+
+The cluster will now start provisioning. You can check status with:
+
+```bash
+kubectl get cluster
+```
+
+You can get a view of the cluster and its resources as they are created by running:
+
+```bash
+clusterctl describe cluster clusterclass-quickstart
+```
+
+To verify the control plane is up:
+
+```bash
+kubectl get kubeadmcontrolplane
+```
+
+You should see an output similar to the below:
+
+```bash
+NAME                            INITIALIZED   API SERVER AVAILABLE   VERSION   REPLICAS   READY   UPDATED   UNAVAILABLE
+clusterclass-quickstart         true                                 v1.21.2   1                  1         1
+```
+
+
+### Upgrade a Cluster using Managed Topology
+
+The Spec.Topology field added to the Cluster object as part of ClusterClass allows changes made on the Cluster to be propogated across all relevant objects. This turns a Kubernetes cluster upgrade into a one touch operation.
+Looking at the cluster we just created we can see the version of the control plane and the other machines is v1.21.2.
+
+```bash
+> kubectl get kubeadmcontrolplane,machines
```

Comment:
what about getting machine deployment as well, so the user can appreciate that upgrades impact both?

Reply:
I'm using machines because machinedeployment doesn't have a version field in its kubectl get output - I think it's fairly clear to the user, but we could try to include a version in the machinedeployment columns or else add some text explaining what the machines are. WDYT?

Reply:
It should be there, otherwise it's a bug: https://github.com/kubernetes-sigs/cluster-api/blob/07c0a4809361927b15cde2747b34142b7c7ead15/api/v1beta1/machinedeployment_types.go#L285